src/System.Sorting/src/InsertionSort.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System; 
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+
+namespace System.Sorting
+{
+    public class InsertionSorting : Sorting
+    {
+        public InsertionSorting() { }
+        internal override void SortInternal<Tkey>(Tkey[] array, int index, int length, IComparer<Tkey> comparer) 
+        {
+
+            //Cheking the input parameters conditions.
+            Contract.Requires(array != null);
```

Comment:
I would use `Debug.Assert` instead, as that's more in line with what we do other places in CoreFX.